java/src/main/java/com/saucelabs/saucebindings/examples/VisualTest.java

```diff
@@ -0,0 +1,47 @@
+package com.saucelabs.saucebindings.examples;
+
+import com.saucelabs.saucebindings.SaucePlatform;
+import com.saucelabs.saucebindings.VisualSession;
+import com.saucelabs.saucebindings.options.SauceOptions;
+import com.saucelabs.saucebindings.options.VisualOptions;
+import org.junit.Test;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.util.Map;
+
+public class VisualTest {
+
+    @Test
+    public void startSession() {
+        // 1. Create Visual Options instance
+        VisualOptions visualOptions = new VisualOptions();
+        visualOptions.setProjectName("My Project")
+                .setViewportSize("1024x768");
+
+        // 2. Use setVisualOptions() to add the visual options; Make sure setName() has a value
+        SauceOptions sauceOptions = SauceOptions.chrome()
+                .setName("My Test Name")
+                .setPlatformName(SaucePlatform.MAC_HIGH_SIERRA)
+                .setBrowserVersion("68")
+                .setVisualOptions(visualOptions)
+                .build();
+
+        // 3. Start the Session with SauceOptions
+        VisualSession session = new VisualSession(sauceOptions);
```

Comment:
Have we considered enforcing the 3 mandatory parameters through the constructor instead of runtime exceptions?
```
VisualSession session = new VisualSession(sauceOptions, "My Test Name", "My Project Name");
```

Reply:
I don't care about project name. "Build Name" is probably what they want, and if they want something different, `setProjectName()` should be sufficient.

Test Name / init value is a little harder because we can't provide a reasonable default.
I did try some implementations that put it in the constructor, but we probably need to be encouraging people to set this regardless: ```sauceOptions.setName(testName.getMethodName())```

Maybe we update examples for TestRunner instead of just language so this always gets shown?

Reply:
@titusfortner I think `projectName` is mandatory. Hence, I'm suggesting to make it as part of a constructor.
How does `.setName("My Test Name")` propogate to visual? In visual there's no concept of test name, rather a `snapshotName`.
Agreed about `init()`
Can you elaborate on why you want to encourage people to `sauceOptions.setName(testName.getMethodName())`?

Reply:
We only need to avoid the default `projectName` in a Code Parameterized paradigm, not in a Build Differentiated paradigm, and I generally don't think that parameterization is the best solution for configuration variations. Regardless, that can be up to the end user to manage if the default doesn't work for them, but that also needs to be set in `VisualOptions` class, not the `VisualSession` class.

`setName()` is for `init()` not for `takeSnapshot()`.

I could be convinced to require `SauceOptions#setName(Name)` in all Sauce tests, not just visual ones. I wish JUnit made it easier to get the method name by default (I'm spoiled in Ruby because RSpec makes it globally accessible from a class variable). Anyway, we should have that discussion prior to Sauce Bindings v2.0.

Reply:
> We only need to avoid the default projectName in a Code Parameterized paradigm, not in a Build Differentiated paradigm

Actually looking at your code below this isn't true either.

Instead of `"Project Name"` it would be `"Sauce Bindings"` (pulled from the property that is getting set the line before; which is only necessary when running locally, since running it on a CI will use whatever the Build name of the CI is). Actually, we probably should discuss which takes precedence separately.